e2e/e2e_test.go

```diff
@@ -229,24 +248,31 @@ func filterNetworkStatus(networkStatuses []nettypes.NetworkStatus, predicate fun
 	return nil
 }
 
-func secondaryIfaceIPValue(pod *core.Pod, netStatus []nettypes.NetworkStatus) string {
+func secondaryIfaceIPValue(pod *core.Pod) (string, error) {
 	podNetStatus, found := pod.Annotations[nettypes.NetworkStatusAnnot]
-	Expect(found).To(BeTrue(), "expected the pod to have a `networks-status` annotation")
-	Expect(json.Unmarshal([]byte(podNetStatus), &netStatus)).To(Succeed())
-	Expect(netStatus).NotTo(BeEmpty())
-	// Check if interface is net1 and if IP is in range
+	if !found {
+		return "", fmt.Errorf("the pod must feature the `networks-status` annotation")
+	}
+
+	var netStatus []nettypes.NetworkStatus
+	if err := json.Unmarshal([]byte(podNetStatus), &netStatus); err != nil {
+		return "", err
+	}
+
 	secondaryInterfaceNetworkStatus := filterNetworkStatus(netStatus, func(status nettypes.NetworkStatus) bool {
 		return status.Interface == "net1"
 	})
-	Expect(secondaryInterfaceNetworkStatus.IPs).NotTo(BeEmpty())
-	secondaryIfaceIP := secondaryInterfaceNetworkStatus.IPs[0]
 
-	return secondaryIfaceIP
+	if len(secondaryInterfaceNetworkStatus.IPs) == 0 {
+		return "", fmt.Errorf("the pod does not have IPs for its secondary interfaces")
+	}
+
+	return secondaryInterfaceNetworkStatus.IPs[0], nil
 }
 
 func inRange(cidr string, ip string) bool {
 	_, cidrRange, err := net.ParseCIDR(cidr)
-	Expect(err).To(BeNil())
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
```

Comment:
I'm in favor of making inRange return `(bool, error)` rather than using this ExpectWithOffset statement - just to be consistent with removing Expect statements in helper functions. Like so:

```
func inRange(cidr string, ip string) bool, error {
	_, cidrRange, err := net.ParseCIDR(cidr)
	if err != nil {
		return false, err
	}
	ipInRangeCandidate := net.ParseIP(ip)

	return cidrRange.Contains(ipInRangeCandidate)
}
```

And earlier in the code, the following can be changed to accomodate this
```
Expect(inRange(ipv4TestRange, secondaryIfaceIP)).To(BeTrue())

becomes...

ipInRange, err = inRange(ipv4TestRange, secondaryIfaceIP)
Expect(ipInRange).To(BeTrue())
Expect(err).To(BeNil())
```

Reply:
I am just trying to avoid returning `(bool, error)` . That return type makes no sense to me; it should be re-written instead to return an error - and the absence of one, means it worked. 

Your thoughts @nicklesimba ?

Reply:
Your approach certainly reads much better. If removing "Expect"-ations from helper functions is a guideline rather than a strict requirement, then I'm okay with your current implementation.

Reply:
For completeness, my suggestion is to re-write this function to:

```golang
func inRange(cidr string, ip string) error {
	_, cidrRange, err := net.ParseCIDR(cidr)
	if err != nil {
		return err
	}
	if cidrRange.Contains(net.ParseIP(ip)) {
            return nil
        }
        return fmt.Errorf("ip [%s] is NOT in range %s", ip, cidr)
}
```

Which will be used in the tests like: 
```golang
// success
Expect(inRange(ipv4TestRange, secondaryIfaceIP)).To(Succeed())

// failure
Expect(inRange(ipv4TestRange, secondaryIfaceIP)).To(HaveOccurred())
```

Reply:
> Your approach certainly reads much better. If removing "Expect"-ations from helper functions is a guideline rather than a strict requirement, then I'm okay with your current implementation.

Yes, it is not a strict requirement. I just feel expectations belong in the test body. 

For instance, the `ExpectWithOffset` exists exactly to improve the readability of expectations on helpers.

Reply:
Done.